src/public/js/rs.js

```
@@ -21,6 +105,94 @@ var RS = (function() {
     }, '');
   }
 
+  // Encodes using systematic BCH Reed-Solomon on Galois Field 256
+  // Message should be passed in as length k block
+  // k specifies message length
+  // n specifies block length (number of symbols transmitted)
+  // n - k is the number of code symbols added.
+  // Encoding procedure has the largest order coefficient on the left
+  // and the smallest on the right. IE: [1,2,3] -> x^2 + 2x + 3
+  function encodeRSBlock(msg, n, k) {
+    remainder = polynomialDiv(genPoly, msg, n, k)
+    if (remainder == null) {
+      return null;
+    }
+    codedMsg = new Uint8Array(n)
+    for (i = 0; i < k; i++) {
+      codedMsg[i] = msg[i];
+    }
+    for (i = k; i < n; i++) {
+      codedMsg[i] = remainder[i - k]
+    }
+  }
+  function decodeRSBlock(codedMsg, n, k) {
+    remainder = polynomialDiv(genPoly, codedMsg, n, k)
+    if (remainder == null) {
+      return null;
+    }
+    errors = false;
+    for (i = 0; i < n-k; i++) {
+      if (remainder[i] != 0) {
+        errors = true
+      }
+    }
+    if (errors) {
+      // TODO ERROR CORRECTION
+      return null
+    } else {
+      msg = new Uint8Array(k)
+      for (i = 0; i < k; i++) {
+        msg[i] = codedMsg[i]
+      }
+      return msg
+    }
+  }
+
+  function polynomialDiv(divisor, dividend, n, k) {
+    if (divisor.length != k || dividend.length != n) {
+      return null;
```

might want to throw an error instead of returning null?